File: module_wan/src/main/java/com/huyingbao/module/wan/ui/login/action/LoginAction.java
package com.huyingbao.module.wan.ui.login.action;

/**
 * Created by liujunfeng on 2019/1/1.
 */
public interface LoginAction {
    String REGISTER = "register";

    /**
     * 注册
     *
     * @param username   用户名
     * @param password   密码
     * @param repassword 密码
     */
    void register(String username, String password, String repassword);

    String LOGIN = "login";

    /**
     * 登录
     *
     * @param username 用户名
     * @param password 密码
     */
    void login(String username, String password);

    /**
     * 发送验证码,60s之后可以重发
     */
    String GET_IDENTIFY = "getIdentify";

    /**
     * 发送验证码,60s之后可以重发
     */
    void getIdentify();
}

File: module_wan_kotlin/src/main/java/com/huyingbao/module/wan/ui/login/action/LoginAction.kt
package com.huyingbao.module.wan.ui.login.action

/**
 * Created by liujunfeng on 2019/1/1.
 */
interface LoginAction {

    /**
     * 注册
     *
     * @param username   用户名
     * @param password   密码
     * @param repassword 密码
     */
    fun register(username: String, password: String, repassword: String)

    /**
     * 登录
     *
     * @param username 用户名
     * @param password 密码
     */
    fun login(username: String, password: String)

    /**
     * 发送验证码,10s之后可以重发
     */
    fun getIdentify()

    companion object {
        const val REGISTER = "register"

        const val LOGIN = "login"

        /**
         * 发送验证码,10s之后可以重发
         */
        const val GET_IDENTIFY = "getIdentify"
    }
}
